executor/merge_join.go

```diff
@@ -109,16 +109,20 @@ func (t *mergeJoinInnerTable) init(ctx context.Context, chk4Reader *chunk.Chunk)
 	return err
 }
 
-func (t *mergeJoinInnerTable) rowsWithSameKey() ([]chunk.Row, error) {
+func (t *mergeJoinInnerTable) rowsWithSameKey() (*chunk.RowContainer, error) {
 	lastResultIdx := len(t.resultQueue) - 1
 	t.resourceQueue = append(t.resourceQueue, t.resultQueue[0:lastResultIdx]...)
 	t.resultQueue = t.resultQueue[lastResultIdx:]
+	li := chunk.NewList(t.reader.base().retFieldTypes, t.curResult.Capacity(), t.curResult.Capacity())
+	rc := chunk.NewRowContainer(li, nil)
```

Comment:
How to trigger spilling?
And we should add unit test for spilling.

Reply:
Spilling test is added in row_container_test.go, and the trigger is now registered at rowsWithSameKeys where a new RowContainer is allocated.